src/ModelEvaluationOfflineStiffness.cpp

```diff
@@ -0,0 +1,282 @@
+        do{
+            auto frame(ModelEvaluationBase<Logger>::getLmtFromInputQueue());
+            if (!frame.data.empty())
+                lmtDataFromQueue_.push_back(frame);
+            else runCondition = false;
+        } while (runCondition);
+
+
+        runCondition = ModelEvaluationBase<Logger>::externalTorquesAvailable();
+        //read external torques
+        do{
+            auto frame(ModelEvaluationBase<Logger>::getExternalTorquesFromInputQueue());
+            if (!frame.data.empty())
+                externalTorquesDataFromQueue_.push_back(frame);
+            else runCondition = false;
+        } while (runCondition);
+
+
+        runCondition = true;
+        //read moment arms data
+        do{
+            vector< ceinms::InputConnectors::FrameType > maFrame;
+            for (unsigned i = 0; i < noDof_; ++i) {
+                auto currentMomentArms(ModelEvaluationBase<Logger>::getMomentArmsFromInputQueue(i));
+                if (!currentMomentArms.data.empty())
+                    maFrame.push_back(currentMomentArms);
+                else runCondition = false;
+            }
+            if (runCondition)
+                maDataFromQueue_.push_back(maFrame);
+        } while (runCondition);
+
+        runCondition=ModelEvaluationBase<Logger>::momentArmDerivativesAvailable();
+        //read moment arm derivatives data
+        do{
+            vector< ceinms::InputConnectors::FrameType > dMaFrame;
+            for (unsigned i = 0; i < noDof_; ++i) {
+                auto currentMomentArmDerivs(ModelEvaluationBase<Logger>::getMomentArmDerivativesFromInputQueue(i));
+                if (!currentMomentArmDerivs.data.empty())
+                    dMaFrame.push_back(currentMomentArmDerivs);
+                else runCondition = false;
+            }
+            if (runCondition)
+                dMaDataFromQueue_.push_back(dMaFrame);
+        } while (runCondition);
+    }
+
+    template <typename NMSmodelT, typename Logger>
+    void ModelEvaluationOfflineStiffness<NMSmodelT, Logger>::initOfflineCurve() {
+
+        auto lmtIt(lmtDataFromQueue_.begin());
+        for (auto emgIt : emgDataFromQueue_) {
+            double emgTime = emgIt.time + globalEmDelay_;
+            subject_.setTime(emgTime);
+            subject_.setEmgs(emgIt.data);
+            if (lmtIt == lmtDataFromQueue_.begin()) subject_.updateActivations();
+            if (lmtIt != lmtDataFromQueue_.end() && TimeCompare::lessEqual(lmtIt->time, emgTime)) {
+                subject_.updateActivations();
+                subject_.setTime(lmtIt->time);
+                subject_.setMuscleTendonLengths(lmtIt->data);
+                subject_.updateFibreLengths_OFFLINEPREP();
+                ++lmtIt;
+            }
+            subject_.pushState();
+        }
+        subject_.updateFibreLengthTraces();
+    }
+
+
+    template <typename NMSmodelT, typename Logger>
+    void ModelEvaluationOfflineStiffness<NMSmodelT, Logger>::operator()() {
```

Comment:
I don't like the idea of not giving users the opportunity to choose that they want to calculate the stiffness, so that the program can check that all the necessary inputs are provided. Anyway, I just realized that the SimulationManager is messed up with a temporary hack because I didn't put the Stiffness option in the config file, so this is actually a good time to find a proper solution :sweat_smile: Are you fine with putting a lot of *if*s in the evaluators, anyway?